en/docs/authentication-and-authorization/secure-web-applications-with-managed-authentication.md

```diff
@@ -0,0 +1,214 @@
+
+### API calls
+
+You can invoke Choreo APIs within the same organization as your web application using the relative path /choreo-apis/<api-suffix>, regardless of whether managed authentication is enabled for the web application or not.
+
+For example, if the API URL is `https://2d9ec1f6-2f04-4127-974f-0a3b20e97af5-dev.e1-us-east-azure.choreoapis.dev/rbln/item-service/api-e04/1.0.0`, the `<api-suffix>` would be `/rbln/item-service/api-e04/1.0.0`. From your Single Page Application, you can invoke the API using `/choreo-apis/rbln/item-service/api-e04/1.0.0` relative path.
+
+If you enable Choreo's Managed Authentication, you don't have to manually add any logic to attach an access token in the API call, as Choreo APIs will accept the cookies set by Choreo's Managed Authentication.
+You can directly invoke the API. For example:
+
+```
+    const response = await fetch('/choreo-apis/<api-suffix>')
+```
+
+If Choreo's Managed Authentication is disabled, you need to ensure that your web application attaches a valid access token to the API call.
+
+
+### Token Refresh
+
+If the access token provided by your configured Identity Provider expires, you must refresh the tokens before successfully calling a Choreo-deployed API. For token refresh to function,  you need to set up the Identity Provider to issue a refresh token in addition to the access token.
+
+To refresh the tokens, you can make a POST request to the `/auth/refresh` endpoint.
+
+For example:
+
+```
+const response = await fetch("/auth/refresh", { method: "POST" })
+```
+
+If the refresh token is valid, it will undergo a refresh, and the system will respond with a 204 No Content status.
+
+However, if the refresh token has expired, the system will respond with a 401 Unauthorized status.
+
+To automate the token refresh process when receiving a 401 unauthorized response from the Choreo API, you can encapsulate the requests with the refresh logic. Below is an example code snippet that wraps GET requests.
+
+```
+    export const performGetWithRetry = async (url) => {
+        try {
+            // API call
+            return await fetch('/choreo-apis/<api-suffix>');
+        } catch (error) {
+            if (error instanceof HttpError && error.status === 401) {
+                //Received 401 Unauthorized response from API GW. The access token may have expired.
+
+                // Try to refresh the token
+                const refreshResponse = await fetch("/auth/refresh", {
+                    method: "POST"
+                });
+
+                const status = refreshResponse.status;
+                if (status === 401) {
+                    // Session has expired (i.e., Refresh token has also expired).
+                    // Redirect to login page
+                    window.location.href = "/auth/login";
+                }
+                if (status !== 204) {
+                    // The STS cannot refresh the token due to a server error.
```

Comment:
Can we omit the term STS ?

Reply:
Resolved with [c5726a1](https://github.com/wso2/docs-choreo-dev/pull/1183/commits/c5726a15bc558bf96b3fb05ab1f27c4838baf301)